spec/invoice_spec.rb

```diff
@@ -0,0 +1,43 @@
+require_relative 'spec_helper'
+require_relative '../lib/invoice'
+require 'csv'
+
+RSpec.describe Invoice do
+  before :each do
+    @time = Time.now.utc.strftime("%m-%d-%Y %H:%M:%S %Z")
+    @i = Invoice.new({
+      :id          => 6,
+      :customer_id    => 7,
+      :merchant_id => 8,
+      :status      => "pending",
+      :created_at  => @time,
+      :updated_at  => @time,
+      }, nil)
```

Comment:
We could update `nil` to be a mock (`double`). So up at the top of your `before (:each)` you could do `@repo = double('invoice repo')`

And then instead of nil, have @repo.